tests/graceful-recovery.md

```diff
@@ -0,0 +1,97 @@
+# Graceful recovery from restarts
```

Comment:
Could we make a separate directory for this test? This will play nicely with the rest of the tests we're adding.